backend/app/core/schemas/user_register.py

```diff
@@ -0,0 +1,39 @@
+from datetime import date
+
+from dateutil.relativedelta import relativedelta
+from pydantic import BaseModel, EmailStr, field_validator
+
+
+class UserRegister(BaseModel):
+	name: str
+	last_name: str
+	date_of_birth: date
+	email: EmailStr
+	password: str
+
+	@field_validator("name", "last_name")
+	def validate_length(cls, v: str) -> str:
+		if len(v) > 0 and len(v) < 30:
+			return v
+		raise ValueError("The length of the field must be between 0 and 30 characters")
+
+	@field_validator("date_of_birth")
+	def validate_date_of_birth(cls, v: date) -> date:
+		if relativedelta(date.today(), v).years > 18 and relativedelta(date.today(), v).years < 140:
```

Comment:
this check is wrong, users that are 18yo are not permitted, but should be
![image](https://github.com/user-attachments/assets/56548df2-65d3-44e6-9ffe-552684215c6f)